packages/backend-ai/src/conversation/factory.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  FunctionOutput,
+  functionOutputKey,
+} from '@aws-amplify/backend-output-schemas';
+import {
+  BackendOutputStorageStrategy,
+  ConstructContainerEntryGenerator,
+  ConstructFactory,
+  ConstructFactoryGetInstanceProps,
+  FunctionResources,
+  GenerateContainerEntryProps,
+  ResourceProvider,
+} from '@aws-amplify/plugin-types';
+import {
+  ConversationHandlerFunction,
+  ConversationHandlerFunctionProps,
+} from '@aws-amplify/ai-constructs/conversation';
+import path from 'path';
+import { CallerDirectoryExtractor } from '@aws-amplify/platform-core';
+
+class ConversationHandlerFunctionGenerator
+  implements ConstructContainerEntryGenerator
+{
+  readonly resourceGroupName = 'conversationHandlerFunction';
+
+  constructor(
+    private readonly props: DefineConversationHandlerFunctionProps,
+    private readonly outputStorageStrategy: BackendOutputStorageStrategy<FunctionOutput>
+  ) {}
+
+  generateContainerEntry = ({ scope }: GenerateContainerEntryProps) => {
+    const conversationHandlerFunction = new ConversationHandlerFunction(
+      scope,
+      this.props.name,
+      this.props
+    );
+    this.storeOutput(this.outputStorageStrategy, conversationHandlerFunction);
+    return conversationHandlerFunction;
+  };
+
+  /**
+   * Append conversation handler to defined functions.
+   * Explicitly defined custom handler is customer's function and should be visible
+   * in the outputs.
+   */
+  private storeOutput = (
+    outputStorageStrategy: BackendOutputStorageStrategy<FunctionOutput>,
+    conversationHandlerFunction: ConversationHandlerFunction
+  ): void => {
+    outputStorageStrategy.appendToBackendOutputList(functionOutputKey, {
+      version: '1',
+      payload: {
+        definedFunctions:
+          conversationHandlerFunction.resources.lambda.functionName,
+      },
+    });
+  };
+}
+
+class ConversationHandlerFunctionFactory
+  implements ConstructFactory<ConversationHandlerFunction>
+{
+  private generator: ConstructContainerEntryGenerator;
+
+  constructor(
+    private readonly props: DefineConversationHandlerFunctionProps,
+    private readonly callerStack: string | undefined
+  ) {}
+
+  getInstance = ({
+    constructContainer,
+    outputStorageStrategy,
+    resourceNameValidator,
+  }: ConstructFactoryGetInstanceProps): ConversationHandlerFunction => {
+    resourceNameValidator?.validate(this.props.name);
```

Comment:
I made the `name` mandatory for now to avoid this. (and force customer to give it a name they can find later).
Defining custom handler is expected to be rather niche scenario, so that friction might be ok and we can improve it later.